sanic/request.py

```diff
@@ -172,13 +188,19 @@ async def respond(
         headers: Optional[Union[Header, Dict[str, str]]] = None,
         content_type: Optional[str] = None,
     ):
+        try:
+            if self.stream is not None and self.stream.response:
+                raise ResponseException("Second respond call is not allowed.")
+        except AttributeError:
+            pass
         # This logic of determining which response to use is subject to change
         if response is None:
             response = (self.stream and self.stream.response) or HTTPResponse(
                 status=status,
                 headers=headers,
                 content_type=content_type,
             )
```

Comment:
Still need to change this not to test for `self.stream and self.stream.response` (thanks to the test you added right before this, it is always `None`).

Reply:
Fixed.